File: Frontend/bookmyplay/src/services/slotService.js
import api from "./api";

export const getSlotsByVenue = (venueId, date) => {
    const url = date ? `/slots/venue/${venueId}?date=${date}` : `/slots/venue/${venueId}`;
    return api.get(url);
};

export const addSlot = (slot) => {
    return api.post("/slots", slot);
};

export const deleteSlot = (id) => {
    return api.delete(`/slots/${id}`);
};

File: Backend/bookmyplay-api/src/main/java/com/bookmyplay/service/SlotService.java
package com.bookmyplay.service;

import com.bookmyplay.entity.Slot;

import java.util.List;

public interface SlotService {

    List<Slot> getSlotsByVenue(Long venueId, java.time.LocalDate date);

    void generateSlotsForVenue(com.bookmyplay.entity.Venue venue);

    void deleteFutureUnbookedSlots(Long venueId);
}